Develop3

SashiFazya opened the pull request.

SashiFazya committed:
Проверка регистрации пользователя

SashiFazya committed:
Проверка логина пользователя

SashiFazya committed:
Проверка переходов из личного кабинета
Проверка переходов в конструкторе
Добавлена возможность выбора браузера

SashiFazya committed:
Добавлен отчет allure

nikolay-977 commented on src/main/java/pageObject/ForgotPasswordPage.java:
```diff
@@ -0,0 +1,27 @@
+package pageObject;
```
Необходимо исправить наименование пакетов в соответствии с правилами нейминга. Пакеты — допустимы только строчные буквы.

nikolay-977 commented on src/main/java/pageObject/MainPage.java:
```diff
@@ -0,0 +1,51 @@
+    private By profileEntranceButton = By.xpath(".//button[text()='Войти в аккаунт']");
+    private By constructBurgerHeader = By.xpath(".//h1[text()='Соберите бургер']");
+
+    public static final By constructorBunTab = By.xpath(".//span[text()='Булки']/parent::div");
```
Необходимо исправить: для веб-элементов не следует использовать модификатор static

nikolay-977 commented on src/main/java/pageObject/RegistrationPage.java:
```diff
@@ -0,0 +1,60 @@
+        driver.findElement(inputFieldPassword).sendKeys(user.getPassword());
+    }
+
+    public void clickAlreadyRegistratedEntranceButton() {
```
Необходимо исправить: При реализации шагов используй аннотацию `@Step("Создание заказа")`. Без аннотации `@Step` шаг не отобразится в Allure отчете. В скобках указывай описание шага. Это сделает отчет Allure информативнее, что позволит сократить время локализации дефекта.

nikolay-977 commented on src/test/java/TestLogin.java:
```diff
@@ -0,0 +1,76 @@
+    @DisplayName("вход через кнопку «Личный кабинет»")
+    public void checkPersonalAccountPageLogin() {
+        mainPage.clickPersonalAccountButton();
+        assertEquals(loginPage.getURL(), driver.getCurrentUrl());
```
Необходимо исправить: Проверки должны быть в конце тестовых методов. Между проверками и после проверок не должно быть дополнительных действий. Иначе в случае фейла первой проверки, последующие действия(шаги и методы) не будут выполнены, это может аффектить другие тесты, что увеличит время локализации дефекта. Если нужны дополнительные проверки, лучше реализовать отдельные кейсы.

SashiFazya committed:
Поправила наименования пакетов.
Избавилась от static для веб-элементов.
Расписала аннотацию Step.
Asserts только в конце тестов.

nikolay-977 commented on src/main/java/api/methods/UserAPIMethods.java:
```diff
@@ -0,0 +1,76 @@
+        return new User(email, password, name);
+    }
+
+    public boolean createUser(User user) {
```
Необходимо исправить: При реализации шагов используй аннотацию `@Step("Создание заказа")`. Без аннотации `@Step` шаг не отобразится в Allure отчете. В скобках указывай описание шага. Это сделает отчет Allure информативнее, что позволит сократить время локализации дефекта.
